test/jdk/java/net/httpclient/HttpRequestNewBuilderTest.java

```diff
@@ -294,53 +306,70 @@ public void testUserDefinedMethod(HttpRequest request) {
     }
 
     @Test(dataProvider = "testRequests")
-    public void testHeaders(HttpRequest request) {
+    public void testAddHeader(HttpRequest request) {
         BiPredicate<String, String> filter = (n, v) -> true;
-        var r = HttpRequest.newBuilder(request, filter).headers("newName", "newValue").build();
 
+        var r = HttpRequest.newBuilder(request, filter).headers("newName", "newValue").build();
         assertEquals(r.headers().firstValue("newName").get(), "newValue");
         assertEquals(r.headers().allValues("newName").size(), 1);
         assertAllOtherElementsEqual(r, request, "headers");
     }
 
     @Test(dataProvider = "testRequests")
     public void testRemoveHeader(HttpRequest request) {
-        BiPredicate<String, String> filter = (n, v) -> !n.equalsIgnoreCase("testName");
+        BiPredicate<String, String> filter = (n, v) -> !n.equalsIgnoreCase("testName1");
 
         var r = HttpRequest.newBuilder(request, filter).build();
-        assertFalse(r.headers().map().containsKey("testName"));
-        assertHeadersEquals(r, request, filter);
+        assertFalse(r.headers().map().containsKey("testName1"));
+        assertEquals(r.headers().map(), HttpHeaders.of(request.headers().map(), filter).map());
     }
 
     @Test(dataProvider = "testRequests")
     public void testRemoveSingleHeaderValue(HttpRequest request) {
         BiPredicate<String, String> filter = (n, v) ->
-                n.equalsIgnoreCase("testName") && !v.equalsIgnoreCase("testValue");
+                n.equalsIgnoreCase("testName1") && !v.equalsIgnoreCase("testValue1");
+
+        var r = HttpRequest.newBuilder(request, filter).build();
+        assertFalse(r.headers().map().containsValue("testValue1"));
+        assertEquals(r.headers().map(), HttpHeaders.of(request.headers().map(), filter).map());
+    }
+
+    @Test(dataProvider = "testRequests")
+    public void testRemoveMultipleHeaders(HttpRequest request) {
+        BiPredicate<String, String> filter = (n, v) ->
+                !(((n.equalsIgnoreCase("testName1") && v.equalsIgnoreCase("testValue1")))
+                || ((n.equalsIgnoreCase("testName2") && v.equalsIgnoreCase("testValue2"))));
```

Comment:
It would probably be much clearer if you used `BiPredicate.negate` and `BiPredicate.or` here to combine the two BiPredicates.

Reply:
`Bipredicate.negate` and `BiPredicate.or` now used instead of boolean operators. See https://github.com/openjdk/jdk/pull/1059/commits/68ea89e1c55735adc9230b8abc07e1df3ffc03ca